scanner/Makefile

```diff
@@ -113,60 +122,221 @@ ifdef CI
+
+# Subject prefix.
+certs-subj := /C=US/ST=North Carolina/L=Raleigh/O=Red Hat, Inc./OU=Stackrox Scanner
+
+# DNS name or CN of the current cert target.
+cert-dns = $(notdir $(@D)).$(certs-namespace)
+
+# Certificates rules
+# ==================
+
+.PHONY: certs clean-certs
+
+certs: $(certs-t)
+
+clean-certs:
+	@echo "+ $@"
+	$(SILENT)rm -rf certs/
+
+# Generate keys.
+#
+.PRECIOUS: $(certs-d)/%/key.pem $(certs-d)/%/cert.pem
+
+$(certs-d)/%/key.pem:
+	@echo "+ $@"
+	$(SILENT)mkdir -p $(@D)
+	$(SILENT) openssl genpkey  \
+	    -algorithm RSA \
+	    -out $@
+
+# Generate root certificate authority.
+#
+$(certs-d)/ca/root.pem: $(certs-d)/ca/key.pem
+	@echo "+ $@"
+	$(SILENT) openssl req \
+	    -new \
+	    -x509 \
+	    -nodes \
+	    -days 398 \
+	    -subj "$(certs-subj)/CN=RootCA" \
+	    -key $< \
+	    -out $@
+
+# Generate a certificate signing request.
+#
+$(certs-d)/%/cert.csr: $(certs-d)/%/key.pem
+	$(SILENT) openssl req \
+	    -new \
+	    -subj "$(certs-subj)/CN=$(cert-dns)" \
+	    -key $< \
+	    -out $@
+
+# Generate a certificate.
+#
+.PRECIOUS: $(certs-d)/%/ca.pem
+$(certs-d)/%/ca.pem: $(certs-d)/ca/root.pem
+	@echo "+ $@"
+	$(SILENT)cp $< $@
+
+$(certs-d)/%/cert.pem: $(certs-d)/%/cert.csr $(certs-d)/%/ca.pem $(certs-d)/ca/key.pem
+	@echo "+ $@"
+	$(SILENT) openssl x509 \
+	    -req \
+	    -days 365 \
+	    -in $< \
+	    -CA $(@D)/ca.pem \
+	    -CAkey $(certs-d)/ca/key.pem \
+	    -CAcreateserial \
+	    -sha256 \
+	    -extfile <(echo "subjectAltName = DNS:$(cert-dns) DNS:localhost DNS:127.0.0.1") \
+	    -out $@
+
+
+# Verify certs (loose sanity check).
+#
+.PRECIOUS: $(certs-d)/%/.verified
+$(certs-d)/%/.verified: $(certs-d)/%/cert.pem
+	$(SILENT) openssl verify \
+	    -CAfile $(@D)/ca.pem \
+	    $<
+	@echo >$@
+
+#########
+## E2E ##
+#########
+
+# E2E vars
+# ========
+
+SCANNER_E2E_TAG       ?= $(TAG)
+SCANNER_E2E_NAMESPACE ?= $(SCANNER_NAMESPACE)
+
+e2e-chart     := e2etests/helmchart
+e2e-namespace := $(SCANNER_E2E_NAMESPACE)
+e2e-tag       := $(SCANNER_E2E_TAG)
+
+e2e-conf-files := db-postgresql.conf \
+                  db-pg_hba.conf
+
+e2e-certs := ca.pem \
+             scanner-v4-key.pem \
+             scanner-v4-cert.pem \
+             scanner-v4-db-key.pem \
+             scanner-v4-db-cert.pem
+
+e2e-files-d      := $(e2e-chart)/files
+e2e-conf-files-t := $(addprefix $(e2e-files-d)/,$(e2e-conf-files))
+e2e-certs-t      := $(addprefix $(e2e-files-d)/,$(e2e-certs))
+
+# E2E rules
+# =========
+
+.PHONY: e2e-deploy clean-e2e
+
+ifeq ($(SILENT),@)
+.SILENT: e2e-deploy $(e2e-conf-files-t)
+endif
+
+e2e-deploy: $(e2e-conf-files-t) $(e2e-certs-t)
+	@echo "+ $@" $(e2e-tag)
+	$(SILENT)-kubectl create namespace $(e2e-namespace)
+	helm upgrade scanner-v4-e2e $(e2e-chart) \
+	    --install \
+	    --namespace $(e2e-namespace) \
+	    --set image.tag="$(e2e-tag)"
+
+clean-e2e:
+	@echo "+ $@"
+	$(SILENT)rm -rf $(e2e-files-d)/*
+
+# DB configuration files.
+#
+$(e2e-files-d)/db-%.conf: image/db/%.conf
```

Comment:
These files will not live here for much longer. The plan is to move them to `image/templates/helm/shared/config-templates/scanner-v4-db/<whatever>.conf.default` https://github.com/stackrox/stackrox/pull/7706/files#diff-fecde9f17cf54d6eb738d58fbc875abb7ba0a190af92127584258f1a5cb22c25

This is to match CentralDB

Reply:
OK.

Reply:
Wait, why don't we keep them inside `scanner/` and copy them over `i/t/helm/shared/config-tempaltes/<name>/` during helm builds?

Also, will they be exactly the same between this (standalone) and that (stackrox)? Otherwise, it might be a good idea to keep the template in a common place, and each build populates/modifies accordingly.

Anyway, for now E2E seems fine and when the other files come in we update.